autoload/OmniSharp/actions/members.vim

```diff
@@ -18,12 +18,66 @@ function! s:StdioFind(Callback) abort
   let opts = {
   \ 'ResponseHandler': function('s:StdioFindRH', [a:Callback])
   \}
-  call OmniSharp#stdio#Request('/currentfilemembersasflat', opts)
+  call OmniSharp#stdio#Request('/v2/codestructure', opts)
 endfunction
 
 function! s:StdioFindRH(Callback, response) abort
   if !a:response.Success | return | endif
-  call a:Callback(OmniSharp#locations#Parse(a:response.Body))
+  call a:Callback(s:ParseCodeStructure(a:response.Body))
+endfunction
+
+function! s:ParseCodeStructure(responseBody) abort
+  let locations = []
+  let filename = expand('%:p')
+  for element in a:responseBody.Elements
+    call s:ParseCodeStructureItemRec(element, filename, locations)
+  endfor
+  return locations
+endfunction
+
+function! s:ParseCodeStructureItemRec(item, filename, locations) abort
+  call add(a:locations, s:ParseCodeStructureItem(a:item, a:filename))
+  let children = get(a:item, 'Children', [])
+  if type(children) == type([])
+    for child in children
+      call s:ParseCodeStructureItemRec(child, a:filename, a:locations)
+    endfor
+  endif
+endfunction
+
+function! s:ParseCodeStructureItem(item, filename) abort
+  return {
+    \ 'filename': a:filename,
+    \ 'lnum':     a:item.Ranges.name.Start.Line,
+    \ 'col':      a:item.Ranges.name.Start.Column,
+    \ 'end_lnum': a:item.Ranges.name.End.Line,
+    \ 'end_col':  a:item.Ranges.name.End.Column - 1,
+    \ 'text':     s:ComputeItemSignature(a:item),
+    \ 'vcol': 1
+  \}
+endfunction
+
+function! s:ComputeItemSignature(item) abort
```

Comment:
On my first test, I hit an error here, as the `namespace` in my file has no `Properties`:

```json
  "Body": {
    "Elements": [
      {
        "Kind": "namespace",
        "Name": "...",
        "DisplayName": "...",
        "Children": [ ... ],
        "Ranges": { ... },
        "Properties": null
      }
    ]
  },
```

I fixed this by adding this block here, at the start of `s:ComputeItemSignature()`:

```vim
  if type(a:item.Properties) != type({})
    return a:item.DisplayName
  endif
```